lib/presentation/widgets/home_presentation/home_drawer.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+
+import '../../../config/routes/router.dart';
+import '../../../core/utils/ui_constants/keys.dart';
+import '../../view/home.dart';
+
+class HomeDrawer extends StatelessWidget {
+  const HomeDrawer({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Drawer(
+      child: ListView(
+        children: [
+          ListTile(key: homeTailKey,
+            title: const Text('Home'),
+            onTap: () {
+              Navigator.pop(context);
+              Navigator.pushNamed(
+                context,
+                MyRouter.homeView,
+              );
+            },
+          ),
+          ListTile(key: nowPlayingTailKey,
+            title: const Text('Now Playing'),
+            onTap: () {
+              Navigator.pop(context);
+              Navigator.pushNamed(
+                context,
+                MyRouter.movieGrid,
+                arguments: MovieCategory.NOW_PLAYING,
+              );
```

Comment:
Here you can use `popAndPushNamed` instead to made the pop first and then the `pushNamed`

Reply:
addressed